pom.xml

```diff
@@ -495,6 +506,97 @@
           </execution>
         </executions>
       </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-antrun-plugin</artifactId>
+        <version>1.7</version>
+        <executions>
+          <!-- Create the config file for artifact which can be used to deploy the artifact.
+               Sets the parents field to system:cdap-etl-batch and system:cdap-etl-realtime with whatever
+               version range is set in the etl.versionRange property.
+               also sets a widget and doc property for each file contained in the widgets and docs directories. -->
+          <execution>
+            <id>create-artifact-config</id>
+            <phase>prepare-package</phase>
+            <configuration>
+              <target>
+                <script language="javascript"> <![CDATA[
+                  importClass(java.io.File);
+                  importClass(java.io.FileWriter);
+                  importClass(java.io.BufferedWriter);
+                  importClass(java.nio.file.Files);
+                  importClass(java.nio.charset.StandardCharsets);
+
+                  // for some reason, project.basedir evaluates to null,
+                  // so main.basedir is set to project.basedir and then used here
+                  var baseDir = project.getProperty("main.basedir");
+                  var targetDir = project.getProperty("project.build.directory");
+                  var artifactId = project.getProperty("project.artifactId");
+                  var version = project.getProperty("project.version");
+
+                  var cfgFile = new File(targetDir, artifactId + "-" + version + ".json");
+                  if (!cfgFile.exists()) {
+                    cfgFile.createNewFile();
+                  }
+
+                  var etlRange = project.getProperty("etl.versionRange");
+                  var config = {
+                    "parents": [ "system:cdap-etl-batch" + etlRange, "system:cdap-etl-realtime" + etlRange ],
```

Comment:
we haven't merged the batch and realtime apps into a single jar, we just got rid of the batch and realtime plugin jars. Instead of having cdap-etl-lib-3.3.0-batch extend cdap-etl-batch and cdap-etl-lib-3.3.0-realtime extend cdap-etl-realtime, we just have cdap-etl-lib-3.3.0 extend both cdap-etl-batch and cdap-etl-realtime.

We could probably combine the etl batch and realtime jars into one as well... but that would be a code change.